src/ruby/puppetserver-lib/puppet/server/compiler.rb

```diff
@@ -8,17 +9,57 @@ def initialize
         set_server_facts
       end
 
+      # Compiles a catalog according to the spec provided from the
+      # request.
+      # @param [Hash] request_data details about the catalog to be compiled
+      # @return [Hash] containing either just the catalog or catalog and logs,
+      #                if capturing logs was enabled
       def compile(request_data)
+        options = request_data['options'] || {}
```

Comment:
With this pushing `convert_java_args_to_ruby` to after options processing, I think this might be setting options to a `Java::ClojureLang::IPersistentMap`, which is probably not what we want? I think this might mean that we should move that `convert_java_args_to_ruby` out to the `master.rb` file and call it there, so we don't run into this issue again...

Reply:
Oh good catch. Yeah maybe I'll add that refactor we mentioned on Justin's PR here.